### PR TITLE
Add feature to import local module

### PR DESCRIPTION
**Pull request by vutoan1245**
There are two ways to import a local Obojobo document:
1. In the Dashboard: `New Module` -> `Import From File`
2. In the Editor: `File` -> `Import from file...`

Resolve: #1126

### CODE REVIEW

**Comment by vutoan1245** [comment]
@FrenjaminBanklin 

The reason there are two versions of the document is that I was lazy and reused old APIs
Step 1. call `APIUtil.createNewDraft` to create a blank module
Step 2. call `APIUtil.postDraft` to replace the blank module with the content of the select file

Not sure if it is a good idea to do it that way. I feel like I should create a new API for this feature.
I am opening to suggestions.

**Comment by zachberry** [comment]
@vutoan1245 I think perhaps we could update /api/drafts/new to accept content, which would then create the new draft with the given content. If none was provided then it would default to the draftTemplate and draftTemplateXML values.

**Comment by keeeeeegan** [comment]
> The UI doesn't make it very clear that importing from file makes a new module instead of overriding the content of the current system. Maybe move the import button to below new and label it "Import New from File"?
> 
> Also, because the download for JSON currently includes staus codes, users who download a file as JSON and then re-upload it receive a confusing error message "Cannot read property 'start' of undefined"

I'm seeing this behavior as well

**Comment by iturgeon** [comment]
I was having a hard time fixing the conflicts so I just made a new commit w/ the changes Toan did (- test updates) here https://github.com/iturgeon/Obojobo/commit/cbebea749786660120cf5fe1433f06615e348ab9

**Comment by iturgeon** [comment]
welp, that's terrible

File: packages/app/obojobo-repository/shared/components/dashboard.jsx [hunk 1]
```diff
@@ -106,6 +107,33 @@ const Dashboard = props => {
...
+	const onChangeFile = event => {
+		const file = event.target.files[0]
+		if (!file) return
+
+		const reader = new FileReader()
+		reader.readAsText(file, 'UTF-8')
+		reader.onload = function(e) {
+			const contents = e.target.result
+
+			APIUtil.createNewDraft(
```
Comment (iturgeon) [inline packages/app/obojobo-repository/shared/components/dashboard.jsx:119]: This should happen via a redux in an action.

File: packages/app/obojobo-repository/shared/components/dashboard.jsx [hunk 2]
```diff
@@ -106,6 +107,33 @@ const Dashboard = props => {
...
+				contents,
+				file.type === 'application/json' ? 'application/json' : 'application/xml'
+			).then(() => {
+				location.reload()
+			})
+		}
+	}
+
+	const importFile = () => {
+		const fileSelector = document.createElement('input')
```
Comment (iturgeon) [inline packages/app/obojobo-repository/shared/components/dashboard.jsx:129]: This should probably also be in an action

File: packages/app/obojobo-repository/shared/components/dashboard.jsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ const Button = require('./button')
 const MultiButton = require('./multi-button')
 const Search = require('./search')
 const ReactModal = require('react-modal')
+const APIUtil = require('obojobo-document-engine/src/scripts/viewer/util/api-util')
```
Comment (iturgeon) [inline packages/app/obojobo-repository/shared/components/dashboard.jsx:15]: I'd prefer not to use the viewer's api-util here, it has a lot more then we need in it

**Comment by iturgeon** [comment]
Ok, this is ready again

**Comment by vutoan1245** [comment]
I really like the changes you added @iturgeon. It is working great.

File: packages/app/obojobo-document-engine/src/scripts/oboeditor/components/page-editor.js [hunk 4]
```diff
@@ -46,10 +46,12 @@ class PageEditor extends React.Component {
 			showPlaceholders: true
 		}
 
+		this.imported = false
```
Comment (iturgeon) [inline packages/app/obojobo-document-engine/src/scripts/oboeditor/components/page-editor.js:49]: Looking at the diff, this may not be needed any more?

Comment (vutoan1245) [reply packages/app/obojobo-document-engine/src/scripts/oboeditor/components/page-editor.js:49]: True

**Comment by keeeeeegan** [comment]
LGTM, tested locally

**Pull request merged**